api/src/test/java/com/epam/pipeline/acl/metadata/MetadataEntityApiServiceTest.java

```diff
@@ -0,0 +1,587 @@
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.TEST_LONG_SET;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.TEST_STRING;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.TEST_STRING_MAP;
+import static com.epam.pipeline.util.CustomAssertions.assertThrows;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+public class MetadataEntityApiServiceTest extends AbstractAclTest {
+
+    private final MetadataClass metadataClass = MetadataCreatorUtils.getMetadataClass();
+    private final MetadataEntityVO metadataEntityVO = MetadataCreatorUtils.getMetadataEntityVO(ID);
+    private final Folder folder = FolderCreatorUtils.getFolder(ID, SIMPLE_USER);
+    private final MetadataEntity metadataEntity = MetadataCreatorUtils.getMetadataEntity(ID, folder);
+    private final PagedResult<List<MetadataEntity>> pagedResult = MetadataCreatorUtils.getPagedResult();
+    private final MetadataFilter metadataFilter = MetadataCreatorUtils.getMetadataFilter(ID);
+    private final MetadataField metadataField = MetadataCreatorUtils.getMetadataField();
+    private final MultipartFile file = new MockMultipartFile(TEST_STRING, TEST_STRING.getBytes());
+    private final InputStream inputStream = new ByteArrayInputStream(TEST_STRING.getBytes());
+
+    private final List<MetadataEntity> metadataEntities = Collections.singletonList(metadataEntity);
+    private final List<MetadataField> metadataFields = Collections.singletonList(metadataField);
+    private final List<MetadataClassDescription> descriptions =
+            Collections.singletonList(MetadataCreatorUtils.getMetadataClassDescription());
+
+    @Autowired
+    private MetadataEntityApiService entityApiService;
+
+    @Autowired
+    private MetadataEntityManager mockMetadataEntityManager;
+
+    @Autowired
+    private MetadataUploadManager mockMetadataUploadManager;
+
+    @Autowired
+    private MetadataDownloadManager mockMetadataDownloadManager;
+
+    @Autowired
+    private EntityManager mockEntityManager;
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldCreateMetadataClassForAdmin() {
+        doReturn(metadataClass).when(mockMetadataEntityManager).createMetadataClass(TEST_STRING);
+
+        assertThat(entityApiService.createMetadataClass(TEST_STRING)).isEqualTo(metadataClass);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyCreateMetadataClassForNotAdmin() {
+        doReturn(metadataClass).when(mockMetadataEntityManager).createMetadataClass(TEST_STRING);
+
+        assertThrows(AccessDeniedException.class, () -> entityApiService.createMetadataClass(TEST_STRING));
+    }
+
```

Comment:
Let's add test for `MetadataEntityApiService.loadAllMetadataClasses` even if there is no acl on the method yet. Let's test that it works for a regular user.